src/client/app/graph/canvas/canvas.service.ts

```diff
@@ -123,6 +128,33 @@ export class CanvasService {
     }
   }
 
+  getCardById(id: string): Card {
+    // TODO(eyuelt): change CollaborativeList to CollaborativeMap
+    let cardsArray = this.cards.asArray();
+    for (let card of cardsArray) {
+      if (card.id === id) {
+        return card;
+      }
+    }
+    return null;
+  }
+
+  connectArrowAndCard(arrow: Arrow, card: Card, arrowConnection: ArrowConnectionType) {
+    if (arrowConnection === ArrowConnectionType.INCOMING) {
+      if (arrow.toCard) {
+        arrow.toCard.incomingArrows.removeValue(arrow);
+      }
+      arrow.toCard = card;
+      card.incomingArrows.push(arrow);
+    } else if (arrowConnection === ArrowConnectionType.OUTGOING) {
```

Comment:
Seems like this branch is never called?

Also, what's the meaning of "INCOMING" vs. "OUTGOING"? Is that enum necessary with the existing split of the arrows to "incoming" and "outgoing"?

Reply:
It's not called right now, but it will be if/when we add the ability to move the base of an arrow.
The enum just specifies how to connect the card to the arrow. I guess I could change this param to be a bool instead of using this enum. What do you think?